app/components/job-tabs.js

```diff
@@ -1,11 +1,24 @@
+import { A } from '@ember/array';
+import { computed } from '@ember/object';
 import { isEmpty } from '@ember/utils';
 import Component from '@ember/component';
 
 export default Component.extend({
-
   tagName: 'div',
   classNames: ['travistab'],
 
+  messagesMaxLevel: computed('request.messages', function () {
```

Comment:
This might be a culprit for badge not showing up 😅 Currently the computed property tracks only the `request.messages` property itself, but not it's content. E.g. the CP will re-calculate on `set('request.mesages', something)`, but it won't react on `get('request.messages').pushObject(message)` - because it doesn't track _content_ changes. In order to make Ember track contents, we should provide it a path:
```suggestion
  messagesMaxLevel: computed('request.messages.@each.level', function () {
```
After this, Ember will start also tracking the `level` property of each array item, the array changes (add, remove, etc.) and the property assignment.

Reply:
aaaah! nice. i didn't know about `@each` in those keys.

Reply:
Yeah, and sometimes you can also find `collection.[].prop` syntax - it's essentially the same, just a shorthand.
But there's a limitation with this key - it only works one level deep, so `collection.@each.prop.another` won't work (it'll throw a warning)